source/Application/Component/Widget/ArticleDetails.php

```diff
@@ -646,9 +631,7 @@ public function showZoomPics()
      */
     public function getZoomPics()
     {
-        $aPicGallery = $this->getPictureGallery();
```

Comment:
Hello Alfred,

first of all I want to thank you for all the work you are doing to improve the OXID framework.

In this case I agree with Anton.
When it comes to debugging, it is really handy to be able to put a watch on the variables.

To talk more in general about structuring functions:
I also prefer to have only one return statement per function and this should be the last line of the function and it should return explicitly a previously assigned variable.
If you return something only in the if condition block, the function implicitly returns NULL, if the condition is not met. I prefer the code to be as explicit as possible.

Maybe this PR should be the topic of a wider discussion, as accepting it, it would implicitly introduce a new coding style into the framework ;-)


Regards


Robert
On Jun 20 2016, at 2:23 pm, Anton Fedurtsya <notifications@github.com> wrote:

In source/Application/Component/Widget/ArticleDetails.php<https://github.com/OXID-eSales/oxideshop_ce/pull/414#discussion_r67680280>:

> @@ -646,9 +631,7 @@ public function showZoomPics()
>       */
>      public function getZoomPics()
>      {
> -        $aPicGallery = $this->getPictureGallery();


This code gives advantage in debugging case. You can easily check whats the results in the variable. Does it really worth to remove the variables?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/OXID-eSales/oxideshop_ce/pull/414/files/83902841d46e417a24275cb2b15adf420b4e781a#r67680280>, or mute the thread<https://github.com/notifications/unsubscribe/AHgO7pwekCpvWhH_0MZtkw_OwjL6Urn4ks5qNoYkgaJpZM4I5om_>.